docs/constants/navigation.js

```diff
@@ -1,574 +1,578 @@
-  starting: sortedStarting,
-  general: sortedGeneral,
-  eas: sortedEas,
-  preview: sortedPreview,
-  featurePreview: sortedFeaturePreview,
-  reference: { ...sortedReference, latest: sortedReference['v' + packageVersion] },
+  starting,
+  general,
+  eas,
+  preview,
+  featurePreview,
+  /** @type {any} */
+  reference: { ...reference, latest: reference[LATEST_VERSION] },
+  generalDirectories,
+  startingDirectories,
+  previewDirectories,
+  easDirectories,
+  featurePreviewDirectories,
   hiddenSections,
   collapsedSections,
 };
+
+// --- MDX methods ---
+
+function makeSection(name, children = []) {
+  // TODO(cedric): refactor node types to match unist
+  return { name, children };
+}
+
+function makeGroup(name, children = [], href = '') {
+  // TODO(cedric): refactor node types to match unist
+  return { name, href, posts: children };
+}
+
+/**
+ * Parse the MDX page and extract the frontmatter/yaml page information.
+ * It will only look for the frontmatter/yaml block in the root nodes.
+ * This requires the `remark-frontmatter` MDX plugin.
+ *
+ * @param {string} file
+ */
+function makePage(file) {
+  const filePath = !path.isAbsolute(file) ? path.resolve(PAGES_DIR, file) : file;
+  const contents = fs.readFileSync(filePath, 'utf-8');
+  const url = pageUrl(filePath);
+  const data = frontmatter(contents).attributes;
+
+  if (!data) {
+    console.error('Page YAML block is unreadable:', file);
+  } else if (!data.title) {
+    console.error('Page does not have a `title`:', file);
+    data.title = '';
+  }
+
+  const result = {
+    // TODO(cedric): refactor name into title
+    name: data.title,
+    // TODO(cedric): refactor href into url
+    href: url,
+  };
+  // TODO(cedric): refactor sidebarTitle into metadata
+  if (data.sidebar_title) {
+    result.sidebarTitle = data.sidebar_title;
+  }
+  // TODO(cedric): refactor sidebarTitle into metadata
+  if (data.hidden) {
+    result.hidden = data.hidden;
+  }
+  return result;
+}
+
+// --- Other helpers ---
+
+/**
+ * Load all pages from a single directory.
+ */
+function pagesFromDir(dir) {
+  return fs
+    .readdirSync(path.resolve(PAGES_DIR, dir), { withFileTypes: true })
+    .filter(entity => entity.isFile())
+    .map(file => makePage(path.join(dir, file.name)));
+}
+
+/**
+ * Create the page url using the absolute file path.
+ */
+function pageUrl(file) {
+  return path
+    .relative(path.resolve(PAGES_DIR, '../'), file)
+    .replace(path.extname(file), '')
+    .replace(/\\/g, '/');
+}
+
+/**
+ * Sort the list of pages alphabetically by either the sidebarTitle or title.
+ */
+function sortAlphabetical(pages) {
+  return pages.sort((a, b) => {
+    const aTitle = a.sidebarTitle || a.name;
+    const bTitle = b.sidebarTitle || b.name;
+    return aTitle.localeCompare(bTitle);
+  });
+}
+
+/**
+ * Sort the list of React Native pages by legacy custom sorting.
+ */
+function sortLegacyReactNative(pages) {
```

Comment:
This part is a bit tricky to rewrite. Some of these files only exists in certain React Native versions. Because we just generate the API structure from the existing files, we have to reuse the old sorting strategy here: list with pages and try to sort if it exists in the list.